refactor: complete migration to React 19

lumirlumir opened the pull request:
<!--
    Thank you for contributing!

    ESLint adheres to the [JS Foundation Code of Conduct](https://eslint.org/conduct).
-->

#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?
<!--
    Please ensure your pull request is ready:

    - Read the pull request guide (https://eslint.org/docs/latest/contribute/pull-requests)
    - Update documentation for this change (if appropriate)
-->

<!--
    The following is required for all pull requests:
-->

#### What changes did you make? (Give an overview)

Working in progress.

#### Related Issues

#### Is there anything you'd like reviewers to focus on?

<!-- markdownlint-disable-file MD004 -->

nzakas reviewed (approved):
LGTM, although I don't know much about React. :smile: Everything appears to work.

Would like another review before merging.

lumirlumir commented on src/playground/components/CodeEditor.js:
```diff
@@ -1,4 +1,4 @@
-import React, { forwardRef, useImperativeHandle, useMemo, useRef } from "react";
+import React, { useImperativeHandle, useMemo, useRef } from "react";
```
This comment isn't related to the changes here, so I'll close it.

lumirlumir commented:
I noticed from this comment (https://github.com/eslint/code-explorer/pull/116#discussion_r2198316439) that the `displayName` property isn't really necessary 😄.

So, I added a new commit https://github.com/eslint/eslint.org/pull/740/commits/d599f775ede9c72234900c249016aac6aa1d52e4.

Also, since the `function` convention is used consistently in other components, I updated it to follow that convention.
